pyscf/mcscf/casci.py

```diff
@@ -283,7 +285,39 @@ def cas_natorb(mc, mo_coeff=None, ci=None, eris=None, sort=False,
         if sort:
             orbsym[ncore:nocc] = orbsym[ncore:nocc][casorb_idx]
         mo_coeff1 = lib.tag_array(mo_coeff1, orbsym=orbsym)
+    else:
+        orbsym = numpy.zeros(nmo, dtype=int)
 
+    # When occupancies of active orbitals equal to 2 or 0, these orbitals
+    # need to be canonicalized along with inactive(core or virtual) orbitals
+    # using general Fock matrix. Because they are strongly coupled with
+    # inactive orbitals, the 0th order Hamiltonian of MRPT methods can be
+    # strongly affected. Numerical uncertainty may be found in the perturbed
+    # correlation energy.
+    # See issue https://github.com/pyscf/pyscf/issues/1041
+    occ2_idx = numpy.where(2 - cas_occ < FRAC_OCC_THRESHOLD)[0]
+    occ0_idx = numpy.where(cas_occ < FRAC_OCC_THRESHOLD)[0]
+    if occ2_idx.size > 0 or occ0_idx.size > 0:
+        fock_ao = mc.get_fock(mo_coeff, ci, eris, casdm1, verbose)
+
+        def _diag_subfock_(idx):
+            c = mo_coeff1[:,idx]
+            fock = reduce(numpy.dot, (c.conj().T, fock_ao, c))
+            w, c = mc._eig(fock, None, None, orbsym[idx])
+            mo_coeff1[:,idx] = mo_coeff1[:,idx].dot(c)
+
+        if occ2_idx.size > 0:
+            log.debug('Active orbitals %s (occs = %s) are canonicalized with core orbitals',
```

Comment:
This is more of a stylistic comment, but it might be safer to use `log.warn` instead of `log.debug`, just so users are more likely to see these comments